core/src/test/java/io/lionweb/serialization/data/SerializedClassifierInstanceTest.java

```diff
@@ -0,0 +1,231 @@
+package io.lionweb.serialization.data;
+
+import static org.junit.Assert.*;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import org.junit.Test;
+
+public class SerializedClassifierInstanceTest {
+
+  private static MetaPointer simpleMetaPointer(String key) {
+    return MetaPointer.get("L", "1", key);
+  }
+
+  @Test
+  public void defaultsAreEmptyAndUnmodifiable() {
+    SerializedClassifierInstance sci = new SerializedClassifierInstance();
+
+    assertNull(sci.getID());
+    assertNull(sci.getClassifier());
+    assertNull(sci.getParentNodeID());
+
+    assertTrue(sci.getProperties().isEmpty());
+    assertTrue(sci.getContainments().isEmpty());
+    assertTrue(sci.getChildren().isEmpty());
+    assertTrue(sci.getReferences().isEmpty());
+    assertTrue(sci.getAnnotations().isEmpty());
+
+    assertThrows(UnsupportedOperationException.class, () -> sci.getProperties().add(null));
```

Comment:
Why would adding a `null` property cause a `UnsupportedOperationException` to be thrown? I'd expect a less generic, more precise exception.

Reply:
That would be the case for each value (I will change the value). The problem is that getProperties return immutable lists, so try to add to them cause this exception to be thrown. The exception is determined by the implementation of the immutable list